controller/api/public/client.go

```diff
@@ -197,19 +211,30 @@ func fromByteStreamToProtocolBuffers(byteStreamContainingMessage *bufio.Reader,
 	return nil
 }
 
-func newClient(apiURL *url.URL, httpClientToUse *http.Client, controlPlaneNamespace string) (APIClient, error) {
+func newClient(apiURL *url.URL, portForward *k8s.PortForward, httpClientToUse *http.Client, controlPlaneNamespace string) (APIClient, error) {
```

Comment:
The new version of this function is a horrifying train wreck to accommodate both callers that include a PortForward and callers that don't. I'm very open to suggestions about how this could be made less upsetting.